docs/basics/primitives.md

```diff
@@ -5,30 +5,195 @@ to describe the set of types that are native to its platform, we leverage and ov
+| InetAddressColumn             | java.net.InetAddress      | inet              |
+| LongColumn                    | scala.Long                | long              |
+| StringColumn                  | java.lang.String          | text              |
+| UUIDColumn                    | java.util.UUID            | uuid              |
+| TimeUUIDColumn                | java.util.UUID            | timeuuid          |
+| ListColumn[Type]              | immutable.List[Type]      | list<type>        |
+| SetColumn[Type]               | immutable.Set[Type]       | set<type>         |
+| MapColumn[Type, Type]         | immutable.Map[Type, Type] | map<type, type>   |
+| CounterColumn                 | scala.Long                | counter           |
+| StaticColumn&lt;type&gt;      | &lt;type&gt;              | type static       |
+
+
+<a id="optional-primitive-columns">Optional primitive columns</a>
+===================================================================
+
+Optional columns allow you to set a column to a ```null``` or a ```None```.
+Use them when you really want something to be optional.
+The outcome is that instead of a ```T``` you get an ```Option[T]``` and you can ```match, fold, flatMap, map``` on a ```None```.
+
+The ```Optional``` part is handled at a DSL level, it's not translated to Cassandra in any way.
+
+| phantom columns               | Java/Scala type                   | Cassandra columns |
+| ---------------               | -------------------------         | ----------------- |
+| OptionalBlobColumn            | Option[java.nio.ByteBuffer]       | blog              |
+| OptionalBigDecimalColumn      | Option[scala.math.BigDecimal]     | decimal           |
+| OptionalBigIntColumn          | Option[scala.math.BigInt]         | varint            |
+| OptionalBooleanColumn         | Option[scala.Boolean]             | boolean           |
+| OptionalDateColumn            | Option[java.util.Date]            | timestamp         |
+| OptionalDateTimeColumn        | Option[org.joda.time.DateTime]    | timestamp         |
+| OptionalDoubleColumn          | Option[scala.Double]              | double            |
+| OptionalEnumColumn            | Option[scala.Enumeration]         | text              |
+| OptionalFloatColumn           | Option[scala.Float]               | float             |
+| OptionalIntColumn             | Option[scala.Int]                 | int               |
+| OptionalInetAddressColumn     | Option[java.net.InetAddress]      | inet              |
+| OptionalLongColumn            | Option[Long]                      | long              |
+| OptionalStringColumn          | Option[java.lang.String]          | text              |
+| OptionalUUIDColumn            | Option[java.util.UUID]            | uuid              |
+| OptionalTimeUUID              | Option[java.util.UUID]            | timeuuid          |
+
+It is also possible to express your entire table logic using just `Col` if you
```

Comment:
@iRhysBradbury I mean we have two type aliases, I'm not sure I'm getting this across properly.